docs/aws/audit/awsiamdeepdive/rules/suspicious_access_to_data.mdx

```diff
@@ -22,6 +22,263 @@ CBP
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Once you have AWS CLI installed and configured, you can start using it to interact with AWS services.
+
+2. To check for suspicious access to data services, you can use the AWS CloudTrail service. CloudTrail records all API calls made in your AWS account, including successful and failed sign-in attempts. You can use the following command to list all CloudTrail trails:
+
+   ```
+   aws cloudtrail describe-trails
+   ```
+
+3. Once you have the name of the trail you want to investigate, you can use the following command to look at the events recorded in that trail:
+
+   ```
+   aws cloudtrail lookup-events --lookup-attributes AttributeKey=EventName,AttributeValue=ConsoleLogin
+   ```
+
+   This command will return a list of all 'ConsoleLogin' events, which represent sign-in attempts to the AWS Management Console.
+
+4. To further investigate suspicious activity, you can filter the events by date, user name, IP address, or other attributes. For example, the following command will return all 'ConsoleLogin' events from a specific user:
+
+   ```
+   aws cloudtrail lookup-events --lookup-attributes AttributeKey=Username,AttributeValue=<username>
+   ```
+
+   Replace `<username>` with the name of the user you want to investigate. This will help you identify any unusual or suspicious access patterns.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. **Setup AWS SDK (Boto3) in Python Environment:**
+   First, you need to set up AWS SDK (Boto3) in your Python environment. You can install it using pip:
+   ```
+   pip install boto3
+   ```
+   After installing boto3, configure it with your user credentials. You can configure it in python script like this:
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       aws_session_token='SESSION_TOKEN',
+   )
+   ```
+2. **Create IAM Client:**
+   Now, create an IAM client using the session created above. This client will be used to interact with AWS IAM.
+   ```python
+   iam = session.client('iam')
+   ```
+3. **Fetch IAM Users:**
+   Fetch all the IAM users and iterate over them to check their access to data services. Here is how you can fetch IAM users:
+   ```python
+   users = iam.list_users()
+   for user in users['Users']:
+       print('User: {0}\nUserID: {1}\nARN: {2}\nCreatedOn: {3}\n'.format(
+           user['UserName'], user['UserId'], user['Arn'], user['CreateDate']))
+   ```
+4. **Check Access to Data Services:**
+   For each user, check their policies and permissions. If a user has access to data services, it might be suspicious. Here is how you can check user policies:
+   ```python
+   user_policies = iam.list_user_policies(UserName=user['UserName'])
+   for policy_name in user_policies['PolicyNames']:
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>